### PR TITLE
Add amp-dynamic-css-classes sample

### PR DESCRIPTION
**Pull request by jmadler**

### CODE REVIEW

**Comment by jmadler** [comment]
Resolves #353

**Comment by jmadler** [comment]
Blocked on https://github.com/ampproject/amphtml/issues/4953

**Comment by jmadler** [comment]
Sebstian and I chatted and I'm going to push the dynamic reordering of social-share buttons into a separate PR and create a simple boolean 'show-if-referred-by' example for this PR.

File: src/20_Components/amp-dynamic-css-classes.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  <span class="if-google">You were referred here or embedded by Google!  I'll add a few extra Google+ buttons</span>
+  <span class="if-ampbyexample">You came here directly!  Cool :)</span>
+  <span class="if-localhost">You came here directly!  Cool :)</span>
+
+  <amp-social-share type="email" layout="relative"></amp-social-share>
+  <amp-social-share type="twitter" layout="relative" class="if-twitter"></amp-social-share>
+  <amp-social-share type="pinterest" layout="relative" class="if-pinterest"></amp-social-share>
+  <amp-social-share type="linkedin" layout="relative" class="if-linkedin"></amp-social-share>
+
+  <!-- #### Viewer -->
```
Comment (sebastianbenz) [inline src/20_Components/amp-dynamic-css-classes.html]: Can you add a test link to the Google viewer in the doc part?

File: src/20_Components/amp-dynamic-css-classes.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  <!--
+    Using these referrer classes can allow you to conditionally include elements.
+  -->
+  <span class="if-pinterest">You were referred here or embedded by Pinterest!  I'll add a few extra "Pin It" buttons</span>
+  <span class="if-twitter">You were referred here or embedded by Twitter!  I'll add a few extra Tweet buttons</span>
+  <span class="if-google">You were referred here or embedded by Google!  I'll add a few extra Google+ buttons</span>
+  <span class="if-ampbyexample">You came here directly!  Cool :)</span>
+  <span class="if-localhost">You came here directly!  Cool :)</span>
+
+  <amp-social-share type="email" layout="relative"></amp-social-share>
```
Comment (sebastianbenz) [inline src/20_Components/amp-dynamic-css-classes.html]: validation error: layout relative is not supported

File: src/20_Components/amp-dynamic-css-classes.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  <script async custom-element="amp-social-share" src="https://cdn.ampproject.org/v0/amp-social-share-0.1.js"></script>
+  <link rel="canonical" href="<%host%>/components/amp-dynamic-css-classes/" >
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <style amp-custom>
```
Comment (sebastianbenz) [inline src/20_Components/amp-dynamic-css-classes.html]: Can you add a comment to the css block explaining the how these rules work.

File: src/20_Components/amp-dynamic-css-classes.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
+<!--
+  #### Introduction
```
Comment (sebastianbenz) [inline src/20_Components/amp-dynamic-css-classes.html]: Can you add a introduction? For formatting reasons, please put the intro text into a separate comment: 

```
<!-- Introduction -->
<!-- Text ... -->
```

File: src/20_Components/amp-dynamic-css-classes.html [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  </style>
+</head>
+<body>
+
+  <!-- You do not need to include a call to the amp-dynamic-css-classes component to initialize the classes on the body element -->
+
+  <!-- #### Referrer -->
+  <!--
+    Using these referrer classes can allow you to conditionally include elements.
+  -->
```
Comment (sebastianbenz) [inline src/20_Components/amp-dynamic-css-classes.html]: Can you wrap the whole block in a `div` to ensure it's visible on mobile.

File: src/20_Components/amp-dynamic-css-classes.html [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  <span class="if-google">You were referred here or embedded by Google!  I'll add a few extra Google+ buttons</span>
+  <span class="if-ampbyexample">You came here directly!  Cool :)</span>
+  <span class="if-localhost">You came here directly!  Cool :)</span>
+
+  <amp-social-share type="email" layout="responsive" height="30" width="30"></amp-social-share>
+  <amp-social-share type="twitter" layout="responsive" height="30" width="30" class="if-twitter"></amp-social-share>
+  <amp-social-share type="pinterest" layout="responsive" height="30" width="30" class="if-pinterest"></amp-social-share>
+  <amp-social-share type="linkedin" layout="responsive" height="30" width="30" class="if-linkedin"></amp-social-share>
+
+  <!-- #### Viewer -->
```
Comment (sebastianbenz) [inline src/20_Components/amp-dynamic-css-classes.html]: Please see my previous comment about adding a link to this page in a viewer

File: src/20_Components/amp-dynamic-css-classes.html (hunk 4 above)
Comment (sebastianbenz) [inline src/20_Components/amp-dynamic-css-classes.html]: Please also add a link to the official docs.

File: src/20_Components/amp-dynamic-css-classes.html (hunk 6 above)
Comment (jmadler) [inline src/20_Components/amp-dynamic-css-classes.html]: Well, the Google AMP Viewer currently redirects to the canonical if viewed from a Desktop UA, so I'm not sure it'd be a meaningful exercise for desktop viewers.  

One thing we could do is provide the link and suggest users use mobile emulation mode in Chrome.  Would that work?

File: src/20_Components/amp-dynamic-css-classes.html (hunk 4 above)
Comment (jmadler) [inline src/20_Components/amp-dynamic-css-classes.html]: Yep!

File: src/20_Components/amp-dynamic-css-classes.html (hunk 5 above)
Comment (jmadler) [inline src/20_Components/amp-dynamic-css-classes.html]: Will do

File: src/20_Components/amp-dynamic-css-classes.html (hunk 3 above)
Comment (jmadler) [inline src/20_Components/amp-dynamic-css-classes.html]: Yep!

**Comment by jmadler** [comment]
Update - PTAL

**Pull request merged**